common/src/desktopMain/kotlin/com/xacalet/minesweeper/common/ui/MainScreen.desktop.kt

```diff
@@ -0,0 +1,43 @@
+package com.xacalet.minesweeper.common.ui
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.padding
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import com.xacalet.minesweeper.common.data.GameRepository
+import com.xacalet.minesweeper.common.model.GameState
+
+// TODO: Join fun Screen in common
+@Composable
+actual fun MainScreen() {
+    // TODO: Provide this as parameter of Board() composable function.
+    val gameRepository = remember { GameRepository() }
+    val gameData by gameRepository.gameData.collectAsState()
+    gameData?.let {
+        GameBoard(it, gameRepository)
+    }
+}
+
+@Composable
+internal actual fun GameStateImage(isPressed: Boolean, gameState: GameState) {
+    val resourcePath: String = if (isPressed) {
+        "smiley_surprised.svg"
+    } else {
+        when (gameState) {
+            GameState.Won -> "smiley_sunglasses.svg"
+            GameState.Lost -> "smiley_dizzy.svg"
+            else -> "smiley_smiling.svg"
+        }
+    }
+
+    Image(
+        modifier = Modifier.padding(6.dp),
+        painter = painterResource(resourcePath),
+        contentDescription = null
+    )
+}
```

Comment:
Most of composables in this project need a platform-specific version because how images are retrieved, but instead of splitting the whole composable, my suggestion is to leave a single composable in the `commonMain` folder and just split the platform-specific parts:
```kotlin
// /../commonMain/.../IconResources.kt
expect object IconResources {

    @Composable
    fun smilingSmiley(): Painter

    @Composable
    fun surprisedSmiley(): Painter

    // ...
}
```
```kotlin
// /../androidMain/.../IconResources.kt
expect object IconResources {

    @Composable
    actual fun smilingSmiley(): Painter = painterResource(R.drawable.ic_smiley_smiling)

    @Composable
    actual fun surprisedSmiley(): Painter = painterResource(R.drawable.ic_smiley_surprised)

    // ...
}
```
```kotlin
// /../desktopMain/.../IconResources.kt
expect object IconResources {

    @Composable
    actual fun smilingSmiley(): Painter = painterResource("images/smiley_smiling.svg")

    @Composable
    actual fun surprisedSmiley(): Painter = painterResource("images/smiley_surprised.svg")

    // ...
}
```
And then just use the desired painter from IconResources:

```kotlin
// /../commonMain/.../MainScreen.kt
@Composable
fun GameStateImage(isPressed: Boolean, gameState: GameState) {
    val painter: Painter = if (isPressed) {
        IconResources.surprisedSmiley()
    } else {
        when (gameState) {
            GameState.Won -> IconResources.sunglassesSmiley()
            GameState.Lost -> IconResources.dizzySmiley()
            else -> IconResources.smilingSmiley()
        }
    }

    Image(
        modifier = Modifier.padding(6.dp),
        painter = painter,
        contentDescription = null
    )
}
```

WDYT?

Reply:
Yes, I really like more your approach. I will update the PR.